### PR TITLE
Fix and improve dr.wrap_ad

### PR DESCRIPTION
**Pull request by Speierers**
This pull request fixes an issue with the `wrap_ad` function when dealing with integer PyTorch tensor arguments. The internal routine responsible for reshaping the gradient tensors was failing as integer tensors don't produce any gradients. 

This fix has been tested and verified to work correctly.

Moreover, this PR adds the ability to specify the `source` and `target` directly passing the module. Here is an example:

```python
import torch
import drjit as dr

dr.wrap_ad(source=torch, target=dr)
def my_func(args):
    ...
```

### CODE REVIEW

File: drjit/router.py [hunk 1]
```diff
@@ -5722,12 +5722,16 @@ def torch_func(x):
         Forward-mode AD isn't currently supported by this operation.
 
     Args:
-        source (str): The AD framework used outside of the wrapped function.
-        target (str): The AD framework used within the wrapped function.
+        source (str, module): The AD framework used outside of the wrapped function.
+        target (str, module): The AD framework used within the wrapped function.
```
Comment (njroussel) [inline drjit/router.py:5726]:
```suggestion
        source (str | module): The AD framework used outside of the wrapped function.
        target (str | module): The AD framework used within the wrapped function.
```

**Review by njroussel — APPROVED** [review]
All good :100:
